frontend/src/components/Stores.vue

```diff
@@ -0,0 +1,197 @@
+          <td class="col-1">
+            <info-or-null :condition="hasSeasons(s)">
+              <ul>
+                <li v-for="season in s.seasons" :key="season.id">
+                  {{ season.season }}
+                </li>
+              </ul>
+            </info-or-null>
+          </td>
+          <td class="col-2">
+            <info-or-null :condition="hasAdditionalInformation(s)">
+              <ul>
+                <li v-if="isValidField(s.additionalInfo.specialField1)">
+                  <p>{{ s.additionalInfo.specialField1 }}</p>
+                </li>
+                <li v-if="isValidField(s.additionalInfo.specialField2)">
+                  <p>{{ s.additionalInfo.specialField2 }}</p>
+                </li>
+              </ul>
+            </info-or-null>
+          </td>
+        </tr>
+        </tbody>
+      </table>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import VPagination from "@hennge/vue3-pagination";
+import "@hennge/vue3-pagination/dist/vue3-pagination.css";
+import InfoOrNull from "./InfoOrNull"
+
+export default {
+  name: 'Stores',
+  components: {
+    VPagination,
+    InfoOrNull
+  },
+  methods: {
+    resetFilter: function () {
+      this.nameFilter = ""
+      this.fetchStoresPage(this.page)
+    },
+    hasSeasons: function (store) {
+      return Array.isArray(store.seasons) && store.seasons.length
+    },
+    isValidField: function (f) {
+      return f && f.length > 0
+    },
+    hasAdditionalInformation: function (store) {
+      return store.additionalInfo && (this.isValidField(store.additionalInfo.specialField1) || this.isValidField(store.additionalInfo.specialField2))
+    },
+    activateReadMore(store) {
+      store.readMoreActivated = !store.readMoreActivated;
+    },
+    updatePageHandler: function (page) {
+      this.fetchStoresPage(page)
+    },
+    fetchStoresPage: function (page) {
+      this.stores = []
+      fetch(`/api/getStores?page=${page - 1}&storeNameFilter=${encodeURIComponent(this.nameFilter)}`)
+          .then(response => response.json())
+          .then(data => {
+            this.stores = data.storesPage;
+            this.numberOfPages = data.totalPages;
+            this.dataIsAvailable = true
+          });
+    },
+    update: function (store) {
+      fetch(`/api/updateStore`, {method: 'PUT', body: JSON.stringify(store)})
+          .then(response => console.log(response.text()))
+    },
+    isDataAvailable: async function () {
+      let result = await (await fetch(`/api/isDataAvailable`)).text() == "true";
+      this.dataIsAvailable = result;
+      return result;
+    }
+  },
+  data() {
+    return {
+      stores: [],
+      page: 1,
+      numberOfPages: 1,
+      nameFilter: "",
+      dataIsAvailable: false
+    }
+  },
+  created() {
+    this.isDataAvailable()
+        .then(result => {
+          if (result) {
+            this.fetchStoresPage(this.page)
+          } else {
+            const interval = setInterval(function () {
```

Comment:
if you make this an arrow function you can drop the `bind(this)` because it inherits the scope